### ISSUE TITLE
Calendar format: Parameter `$srfgFirstDayOfWeek` broken?

### ISSUE THREAD

**Issue by kghbln** [SemanticMediaWiki/SemanticResultFormats#138]
Though I am setting `$srfgFirstDayOfWeek = 'monday';` I am getting a [warning](https://github.com/SemanticMediaWiki/SemanticResultFormats/blob/6a38b2337f19c1963c1a1bc0af028c0da3a8cabd/formats/calendar/SRF_Calendar.php#L380) and it is obviously being [ignored as the example shows](https://www.semantic-mediawiki.org/wiki/Help:Calendar_format#Example). Perhaps this can be turned from a configuration parameter to a result parameter as it is done for the 'eventcalendar' format.

Code starts at [this spot](https://github.com/SemanticMediaWiki/SemanticResultFormats/blob/6a38b2337f19c1963c1a1bc0af028c0da3a8cabd/formats/calendar/SRF_Calendar.php#L362).

**Comment by kghbln** [comment]
@s7eph4n If I am not mistaken you originally authored this configuration parameter. So you might be able to have a look. Thanks and cheers

**Comment by kghbln** [comment]
I will try `$srfgFirstDayOfWeek = 'Monday';` instead of `$srfgFirstDayOfWeek = 'monday';`. It seems that weekday need to have a capital letter in any case. I will reopen in case this does not help.

**Comment by s7eph4n** [comment]
Sorry, I totally missed this.

`$srfgFirstDayOfWeek` is set in [SRF_Calendar.php](https://github.com/SemanticMediaWiki/SemanticResultFormats/blob/master/formats/calendar/SRF_Calendar.php#L374).

It can be either a number (1 = Sunday, 7 = Saturday), or the name of the day in the wiki language. Since in English daynames are capitalized, "Monday" should work, whereas "monday" probably won't.

**Comment by kghbln** [comment]
> "Monday" should work, whereas "monday" probably won't.

Yeah, thanks for confirming. Just reconfigured and it is working. Also fixed the [docu](https://www.semantic-mediawiki.org/w/index.php?title=Help:Calendar_format&curid=392&diff=45705&oldid=42414)

**Comment by kghbln** [comment]
Do not understand me wrong, but this setting really goes me gewaltig auf den Sack - for years now. I now suspect that this is dependent on the locale of the server. I really have not clue when PHP expects what. Now I set this to Monday and the error goes away. Please change this to numbers, please, if possible.

**Comment by kghbln** [comment]
No it does not fix to set to "Monday". Now I am being spammed by cron about this problem. I have no clue, thus I commented out this configuration setting.

**Comment by thomasramm** [comment]
i run into the same problem, and find the solution.
actually the extension try to translate the weekdays into the user language. But this depends on Individual User Setting! Not on Server Setting.
If i set the value in LocalSettings.php to 'Montag' this works only for my Users with Language=German,
If i set the value in LocalSettings.php to 'Monday' this works only for my Users with Language=English.

**Reason**
On SRF_Calender.php the array "$weekDayNames" is filled with weekdays in UserSetting-Language.
i add print_r($weekDayNames); to the code to have a look into this array.
if i set my user language to German i get 
Array ( [1] => Sonntag [2] => Montag [3] => Dienstag [4] => Mittwoch [5] => Donnerstag [6] => Freitag [7] => Samstag ) 
if i set my user language to English i get 
Array ( [1] => Sunday [2] => Monday [3] => Tuesday [4] => Wednesday [5] => Thursday [6] => Friday [7] => Saturday )

But then it try to find the correct value based on the value you set in LocalSettings.php
So the Solution is to translate $srfgFirstDayOfWeek also to the UserLanguage and use the english value in LocalSettings as base to find the correct translation.

**Solution''
The Code needs to change from `$firstDayOfWeek = array_search($srfgFirstDayOfWeek, $weekDayNames );`
to (Line ~393) `$firstDayOfWeek = array_search( wfMessage($srfgFirstDayOfWeek)->text(), $weekDayNames );`

And in LocalSettings.php you have to use the english value, e.g.: $srfgFirstDayOfWeek = 'Monday';

This Works on my Wiki

Also the documentation has to be updated if you adapt this solution. On https://www.semantic-mediawiki.org/wiki/Help:Ergebnisformat_calendar. and all the translations.

**Comment by krabina** [comment]
thank you @thomasramm! I updated the english docs (only)

**Comment by JeroenDeDauw** [comment]
It took 9 years to fix this bug? 😆

**Comment by kghbln** [comment]
Man, it was a complicated one. Don't you see? 😆
